blocks/library/reusable-block/index.php

```diff
@@ -0,0 +1,44 @@
+<?php
+/**
+ * Server-side rendering of the `core/reusable-block` block.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Renders the `core/reusable-block` block on server.
+ *
+ * @param array $attributes The block attributes.
+ *
+ * @return string Rendered HTML of the referenced block.
+ */
+function gutenberg_render_block_core_reusable_block( $attributes ) {
+	$reusable_blocks = get_posts( array(
+		'post_type' => 'wp_block',
+		'name'      => $attributes['ref'],
+	) );
+
+	$reusable_block = array_shift( $reusable_blocks );
+	if ( ! $reusable_block ) {
+		return '';
+	}
+
+	$blocks = gutenberg_parse_blocks( $reusable_block->post_content );
+
+	$block = array_shift( $blocks );
+	if ( ! $block ) {
+		return '';
+	}
+
+	return gutenberg_render_block( $block );
+}
+
+register_block_type( 'core/reusable-block', array(
```

Comment:
All the other terms we can more easily change, like internal variables, actions, and functions. But this we'll have to support once we announce.

I am not super comfortable with the name. A couple ideas:

* `core/global-block`
* `core/saved-block` -> to match the inserter tab.
* or maybe this is an exception and is saved like `global-block` without the core suffix. Not to keen on an exception here, but it might warrant it (also of note, we have `core-embed` as a namespace.

For reference, the CPT is now called `wp-block`.

cc @aduth @youknowriad @mcsf @pento for naming thoughts

Reply:
I'm not a native speaker, so if `reusable-block` is clearest I'm fine with sticking with it, just want to make sure we are ok with it.

Reply:
Thinking a bit more, it seems the most clear and consistent with the `wp-block` post type is to call this `core/block` which maps to `wp:block` and `wp:core/block`. Thoughts?

Reply:
My only hesitation with _global block_ is that _global_ is a fairly technical term and that using it in the UI might limit how approachable this feature is to a wider audience. 

For technical usages though, I'm happy with using a more precise term. `core/block` matching up with `wp-block` is very appealing.

Reply:
_Saved block_ might speak to users more. Further, though I'm not a huge fan, there are precedents out there to justify _custom blocks_. Note that I can see _reusable blocks_ working, if visual context and copy correctly introduce the concept.

Playing devil's advocate, _saved_ and _custom blocks_ may sound confusing ("Aren't I already making my own block when I add a "Paragraph" and type something in? Isn't it custom? Isn't it being saved?"). Those terms stress a different aspect from _reusable_, with _reusable_ arguably stressing the user-facing value. (Mind you, a similar case can be made against _reusable_: "This Paragraph, or this Author block, aren't they reusable?")

Reply:
I'm going to go ahead and rename the block identifier to `core/block` so that we can move ahead with this PR. This makes sense to me since, just as `<!-- wp:image -->` inserts an image, `<!-- wp:block -->` inserts a `wp_block`.

Let's move the discussion about the name of the overall feature to a seperate issue—for what it's worth, I asked my wife (a regular WordPress user) what she thought and she said that _custom block_ makes the most sense 🙂